### PR TITLE
Warn for region header mismatches

### PR DESCRIPTION
**Pull request by ievans3024**
This PR addresses #7 and replaces `assert` statements with logging at the warning level.

Additionally, it adds a `logger` module which supplies `log` with a stdout stream handler attached set to the warning level.

I noticed that some logging verbosity arguments are in `cli.py` but the stuff in `cli.py` doesn't appear to be actively used yet. I have a [gist](https://gist.github.com/ievans3024/4bbc63643ac197135a308c36220f9928) that demonstrates using `--verbose/-v` as a counter to determine log level, that could be useful, but is outside the scope of this PR.

### CODE REVIEW

**Comment by MestreLion** [comment]
Sorry for taking so long to review this! I'll add a few notes on code...

File: mcworldlib/region.py [hunk 1]
```diff
@@ -139,18 +138,20 @@ def parse(cls, buff):
...
-                f'Length mismatch for region {self.pos} in chunk {pos}:' \
-                f' region header declares {sector_count} {SECTOR_BYTES}-byte sectors,' \
-                f' but chunk data required {chunk.sector_count}.'
+            if chunk.sector_count != sector_count:
+                if chunk.sector_count + 1 != sector_count:
+                    log.warning(
+                        f'Length mismatch for region {self.pos} in chunk {pos}:'
+                        f' region header declares {sector_count} {SECTOR_BYTES}-byte sectors,'
+                        f' but chunk data required {chunk.sector_count}.'
+                    )
```
Comment (MestreLion) [inline mcworldlib/region.py:154]: Great! It really needed to replace this assert with a proper logging! But why change the original condition  with double nested `if`s? I believe a single `if not(chunk.sector_count <= sector_count <= chunk.sector_count + 1): log.warning(...` will do the trick

File: mcworldlib/logger.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import logging
+
+stdout_handler = logging.StreamHandler()
+stdout_handler.setLevel(logging.WARNING)
+log = logging.getLogger(__name__)
+log.addHandler(stdout_handler)
```
Comment (MestreLion) [inline mcworldlib/logger.py:6]: I have a few issues with this approach:
- When you setup logging like this, moving `log = logging.getLogger(__name__)` to a dedicated module, `__name__` will always be `logger`, which kind of defeats the purpose. Each module should have their own `log` object named after themselves.
- "package-level" logging configuration _could_ be performed on `__init__.py`, but... a well-behaved library should not do _any_ logging configuration on its own, this is for the clients to do. Also, by default logging is _already_ set with a `StreamHandler` at `WARNING` level, so there's really no need for this module.

**Comment by MestreLion** [comment]
About `cli.py` - Yes, that is just a stub for now. I plan to eventually add a `main()` there so you can use the library on command line. But it already have verbosity handling: by default it runs with `INFO` (as opposed to logging's WARNING default), swtiches to `DEBUG` on `--verbose` and back to `WARNING` on `--quiet`.

The code will be somewhat like this:

```python
def main(argv=None):
    args = parse_args(argv)
    logging.basicConfig(level=args.loglevel,
                        format='%(levelname)-5.5s: %(message)s')
    log.debug(args)

    # Program go here...


if __name__ == '__main__':
    try:
        sys.exit(main(sys.argv[1:]))
    except Exception as e:
        log.critical(e, exc_info=True)
        sys.exit(1)
    except KeyboardInterrupt:
        sys.exit(2)  # signal.SIGINT.value
```

File: mcworldlib/region.py (hunk 1 above)
Comment (ievans3024) [reply mcworldlib/region.py:154]: I believe the main reason was that I wanted to ensure that this warning would be printed for mismatches of any variety, not just if `chunk.sector_count > sector_count`.

My thinking was that you can't chain general inequality together and be guaranteed the expected result in every configuration, but I may be wrong on that.

However, even given that, there are now a couple of conditional flaws I'm seeing here, for example, the previous assertion would have thrown an error if `sector_count == chunk.sector_count + 1` but my code would not.

At the very least, I'll need to fix that much. If you require that the conditional match the original code, then I can make it so.

File: mcworldlib/logger.py (hunk 2 above)
Comment (ievans3024) [reply mcworldlib/logger.py:6]: That makes sense, I wasn't really thinking about that. I will go ahead and undo the logging changes.

File: mcworldlib/region.py (hunk 1 above)
Comment (MestreLion) [reply mcworldlib/region.py:154]: The idea is to allow `sector_count` to be either `chunk.sector_count` or `chunk.sector_count + 1`; any other value will emit a warning. I think `chunk.sector_count <= sector_count <= chunk.sector_count + 1` covers that, but there are my ways to write that (`if sector_count not in (chunk.sector_count, chunk.sector_count+1)` would be another, or using `range()`. Pick the one you think you prefer and think it's most readable. No, the original assertion would not throw error if `sector_count == chunk.sector_count + 1`.

File: mcworldlib/logger.py (hunk 2 above)
Comment (MestreLion) [reply mcworldlib/logger.py:6]: No problem! Once again, thanks for contributing!

**Pull request merged**